From c25945e35bfa558d9b21b05a49f74e52b317fd7f Mon Sep 17 00:00:00 2001
From: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Date: Tue, 14 Nov 2023 02:12:50 -0800
Subject: [PATCH 061/513] test/security: add more algos to combined tests

Add following algos to combined mode cases. These are intended primarily
for TLS record combined mode tests.

- AES256-CBC
- SHA256-HMAC
- SHA1-HMAC [20B]

Signed-off-by: Anoob Joseph <anoobj@marvell.com>
Signed-off-by: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Change-Id: Ia0507d01001fe029d145c17ec0e5b2152d2ebfd2
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/115884
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/117489
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
---
 app/test/test_security_proto.h | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/app/test/test_security_proto.h b/app/test/test_security_proto.h
index 6797e8b8c2f5f..efa023b99d8cd 100644
--- a/app/test/test_security_proto.h
+++ b/app/test/test_security_proto.h
@@ -70,6 +70,12 @@ static const struct crypto_param cipher_list[] = {
 		.key_length = 16,
 		.iv_length = 16,
 	},
+	{
+		.type = RTE_CRYPTO_SYM_XFORM_CIPHER,
+		.alg.cipher =  RTE_CRYPTO_CIPHER_AES_CBC,
+		.key_length = 32,
+		.iv_length = 16,
+	},
 	{
 		.type = RTE_CRYPTO_SYM_XFORM_CIPHER,
 		.alg.cipher =  RTE_CRYPTO_CIPHER_AES_CTR,
@@ -107,12 +113,24 @@ static const struct crypto_param auth_list[] = {
 		.key_length = 20,
 		.digest_length = 12,
 	},
+	{
+		.type = RTE_CRYPTO_SYM_XFORM_AUTH,
+		.alg.auth =  RTE_CRYPTO_AUTH_SHA1_HMAC,
+		.key_length = 20,
+		.digest_length = 20,
+	},
 	{
 		.type = RTE_CRYPTO_SYM_XFORM_AUTH,
 		.alg.auth =  RTE_CRYPTO_AUTH_SHA256_HMAC,
 		.key_length = 32,
 		.digest_length = 16,
 	},
+	{
+		.type = RTE_CRYPTO_SYM_XFORM_AUTH,
+		.alg.auth =  RTE_CRYPTO_AUTH_SHA256_HMAC,
+		.key_length = 32,
+		.digest_length = 32,
+	},
 	{
 		.type = RTE_CRYPTO_SYM_XFORM_AUTH,
 		.alg.auth =  RTE_CRYPTO_AUTH_SHA384_HMAC,
-- 
2.25.1

